soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -312,17 +312,12 @@ static void CreateWothHint(uint8_t* remainingDungeonWothHints) {
     Location(hintedLocation)->SetAsHinted();
     uint32_t gossipStone = RandomElement(gossipStoneLocations);
 
-    // form hint text
-    Text locationText;
     if (Location(hintedLocation)->IsDungeon()) {
         *remainingDungeonWothHints -= 1;
-        uint32_t parentRegion = Location(hintedLocation)->GetParentRegionKey();
-        locationText = AreaTable(parentRegion)->GetHint().GetText();
-
-    } else {
-        uint32_t parentRegion = Location(hintedLocation)->GetParentRegionKey();
-        locationText = GetHintRegion(parentRegion)->GetHint().GetText();
     }
+
+    // form hint text
+    Text locationText = GetHintRegion(Location(hintedLocation)->GetParentRegionKey())->GetHint().GetText();
```

Comment:
so if i'm reading this correctly
```cpp
AreaTable(parentRegion)->GetHint().GetText();
```
and
```cpp
GetHintRegion(parentRegion)->GetHint().GetText();
```

are functionally equivalent, which is why this can be simplified?

Reply:
`AreaTable()` will get the direct area of the region, and if that region hint is "No Hint", then "No Hint" gets used, which is not what we want.

`GetHintRegion()` will check if the region has a valid hint, and if not, search the exits region until it can find a hint. This is how interiors work so that a randomized interior will use the connected region as its hint.

Boss heart containers are now set as "No Hint" so the reverse search can be performed.

Previously if an item was marked as dungeon, then it was guaranteed to have a hint, then there was no need to worry about the reverse search, but thats not the case with Boss shuffle now.